ml_peg/app/utils/build_components.py

```diff
@@ -492,7 +495,42 @@ def build_test_layout(
         column_widths=column_widths,
         thresholds=thresholds,
     )
-    if metric_weights:
+    if threshold_controls and metric_weights:
+        # Combine threshold and weight panels in a single card while trimming the extra
+        # <Br/> injected at the top of the weight component so the boundary box hugs
+        # both controls.
+        weight_children = metric_weights.children
+        if isinstance(weight_children, tuple | list):
+            weight_children = list(weight_children)
+        elif weight_children is None:
+            weight_children = []
+        else:
+            weight_children = [weight_children]
+        if weight_children and isinstance(weight_children[0], Br):
+            weight_children = weight_children[1:]
+        compact_weights = Div(weight_children)
```

Comment:
Does `build_weight_components` not always return `Div(layout)`, where

```
    layout = [
        Br(),
        container,
        Store(
            id=f"{table.id}-weight-store",
            storage_type="session",
            data=weights,
        ),
    ]
``` 

This seems to be trying to make it a lot more general than it needs to be? Or am I missing something?